src/configureWorkspace/configureCompose.ts

```diff
@@ -0,0 +1,109 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from "path";
+import * as vscode from "vscode";
+import { IActionContext } from 'vscode-azureextensionui';
+import { dockerComposeHeader } from "../constants";
+import { ext } from "../extensionVariables";
+import { localize } from "../localize";
+import { captureCancelStep } from '../utils/captureCancelStep';
+import { DockerfileInfo, parseDockerfile } from "../utils/dockerfileUtils";
+import { generateUniqueName } from '../utils/generateUniqueName';
+import { Item, quickPickDockerFileItems } from "../utils/quickPickFile";
+import { quickPickWorkspaceFolder } from "../utils/quickPickWorkspaceFolder";
+import { getComposePorts } from "./configure";
+import { ConfigureTelemetryCancelStep } from './configUtils';
+import { generateNonConflictFileNameWithPrompt, ScaffoldFile, writeFiles } from "./scaffolding";
+
+export interface ComposeScaffoldContext extends DockerfileInfo {
+    serviceName: string
+}
+
+export async function configureCompose(context: IActionContext): Promise<void> {
+
+    function captureStep<TReturn, TPrompt extends (...args: []) => Promise<TReturn>>(step: ConfigureTelemetryCancelStep, prompt: TPrompt): TPrompt {
+        return captureCancelStep(step, context.telemetry.properties, prompt);
+    }
+
+    const serviceNames: string[] = [];
+    let hasDuplicateService: boolean = false;
+    function AddToServiceNames(serviceName: string): void {
+        if (serviceNames.includes(serviceName)) {
+            hasDuplicateService = true;
+        } else {
+            serviceNames.push(serviceName);
+        }
+    }
+
+    const rootFolder: vscode.WorkspaceFolder = await captureStep('folder', promptForFolder)();
+    let composeFile: ScaffoldFile;
+    // Get list of dockerfiles (services) to add to compose
+    const dockerFiles: Item[] = await getDockerFilesInWorkspace(context, rootFolder);
+
+    // Add the services to docker-compose.yaml
+    if (dockerFiles) {
+        const composeScaffoldContexts: ComposeScaffoldContext[] = [];
+        await Promise.all(
+            dockerFiles.map(async dockerfile => {
+                const serviceName = path.basename(path.dirname(dockerfile.absoluteFilePath));
+                const dockerfileInfo: DockerfileInfo = await parseDockerfile(rootFolder.uri.fsPath, dockerfile.absoluteFilePath);
+                AddToServiceNames(serviceName);
+                composeScaffoldContexts.push({
+                    ...dockerfileInfo,
+                    serviceName: serviceName,
+                });
+            })
+        );
+        if (hasDuplicateService) {
```

Comment:
If we already have a list of service names, why wait until now to update `composeScaffoldContexts` with a unique one? Why not make it unique before pushing into `composeScaffoldContexts`?

Reply:
At this point we don't have all the existing names. For examples if the input has (web1, web1, web2, web1.1) then while processing the second web1, it will come up with a unique name of web1.1, which is an actual servicename that wasn't processed yet and then web1.1 will be replaced with web1.2.